src/Generics/SOP/Constraint.hs

```diff
@@ -42,6 +42,24 @@ import Generics.SOP.Sing
 --
 -- means that 'f' can assume that all elements of the n-ary
 -- product satisfy 'Eq'.
+
+-- Note on superclasses: ghc cannot deduce superclasses from 'All'
+-- constraints.
+-- You might expect the following to compile
+-- > class (Eq a) => MyClass a
+-- >
+-- > foo :: (All Eq xs) => NP f xs -> z
+-- > foo = [..]
+-- >
+-- > bar :: (All MyClass xs) => NP f xs -> x
+-- > bar = foo
+-- but it will fail with an error saying that it was unable to
+-- deduce the class constraint 'AllF Eq xs' (or similar) in the
+-- definition of 'bar'.
+-- In cases like this you can use 'Dict' from 'Generics.SOP.Dict'
```

Comment:
modules are references with double quotes: `"Generics.Sop.Dict"`

Reply:
Writing full paths for `Dict` makes it a link too

```
-- In cases like this you can use 'Generics.SOP.Dict.Dict' from "Generics.SOP.Dict"
```

![screenshot from 2018-08-08 13-41-41](https://user-images.githubusercontent.com/51087/43832656-ce3a5278-9b10-11e8-8936-e70cb269150b.png)